### PR TITLE
feat(edit-vid2): Phase 2 - timeline subtitle clip drag/move/resize and detail form

### PR DESCRIPTION
**Pull request by u7chan**
## Issues

- Refs #1068

## Why

Phase 2 of the track-based subtitle editor migration. After sorting and pinning the export panel in Phase 1, we now need to make subtitle clips interactive on the timeline and replace the scrolled list with a detail form for the selected clip.

## Summary

Add drag-to-move and edge-drag-to-resize to `TimelineBar` subtitle clips, replace the right-panel subtitle list with a selected-item detail form, and auto-select newly created subtitles.

## Changes

- Extend `TimelineBar` to render subtitle clips with selection state and resize handles
- Implement `mousedown`/`mousemove`/`mouseup` drag handlers for clip body (move) and edges (resize start/end)
- Add `selectedSubtitleId` state to `EditorPage` and pass it through `TimelineBar`
- Replace the right-panel scrollable subtitle list with `SubtitleDetailForm` for the selected subtitle
- Auto-select a newly added subtitle after the creation PATCH succeeds
- Deselect the current subtitle when clicking empty timeline area
- Update `addSubtitle` E2E helper for the single-detail-form UI
- Add E2E tests for clip selection, body drag, and right-edge resize

## Verification

- `bun run format:check` - passed
- `bun run lint` - passed
- `bun test` - 57 pass, 0 fail
- `bun run build` - passed

## Details

- Clips are constrained to `[0, duration]` and maintain their duration when dragged against the boundaries.
- Drag/resize uses global mouse listeners that are cleaned up on `mouseup`, matching the issue's guidance to avoid `useEffect`.
- The right panel keeps `ExportPanel` fixed at the bottom from Phase 1; the upper area now shows either the detail form or an empty-state placeholder.

### CODE REVIEW

File: projects/edit-vid2/src/client/features/editor/page.tsx [hunk 1]
```diff
@@ -765,19 +812,91 @@ function TimelineBar({
...
+  }
+
+  const commitItemChange = (id: string, nextStart: number, nextEnd: number) => {
+    const item = subtitleItems.find((s) => s.id === id)
+    if (!item) return
+    const minLength = 0.1
+    const sourceStart = Math.max(0, Math.min(roundTime(nextStart), duration - minLength))
+    const sourceEnd = Math.max(sourceStart + minLength, Math.min(roundTime(nextEnd), duration))
+    if (sourceStart === item.sourceStart && sourceEnd === item.sourceEnd) return
+    onChange({ ...item, sourceStart, sourceEnd })
```
Comment (u7chan) [inline projects/edit-vid2/src/client/features/editor/page.tsx]: [must] ⚠️ This calls `onChange` on every `mousemove`, which sends a PATCH for each intermediate drag position. Those requests can be in flight at the same time, and the project endpoint overwrites `timelineState` with whichever request arrives last, so an older intermediate position can persist after the user releases the mouse. Please keep drag state local and persist once on `mouseup` (or otherwise serialize/cancel pending writes) so the stored subtitle reflects the final pointer position.

AIレビュー補助（Codex / gpt-5.5）によるコメントです 🤖

**Review by u7chan — COMMENTED** [review]
AIレビュー補助（Codex / gpt-5.5）によるレビューです 🤖

File: projects/edit-vid2/src/client/features/editor/page.tsx (hunk 1 above)
Comment (u7chan) [reply projects/edit-vid2/src/client/features/editor/page.tsx]: Fixed in the latest commit. Dragging now updates only local preview state (`dragPreview`) during `mousemove` and persists the final position once on `mouseup` via `onChange`. This avoids overlapping PATCH requests and ensures the stored subtitle reflects the final pointer position.

Comment (u7chan) [reply projects/edit-vid2/src/client/features/editor/page.tsx]: Confirmed fixed in the latest head. `mousemove` now updates only local preview state, and the final subtitle range is persisted once on `mouseup`, so the overlapping PATCH race from the previous implementation is avoided.

AIレビュー補助（Codex / gpt-5.5）による再チェックコメントです 🤖
